Assets/Tests/InputSystem/Plugins/InputForUITests.cs

```diff
@@ -127,78 +110,389 @@ public void TouchIsPressedAndMouseEventsAreIgnored()
         });
     }
 
+    void PressUpdateReleaseUpdate(ButtonControl button)
+    {
+        Press(button);
+        Update();
+        Release(button);
+        Update();
+    }
+
+    #region UI_Input_Actions
+    // Test all default UI actions, and sure that InputSystemProvider works with and without project-wide actions asset
+    // so that there is no impact in receiving the necessary input events for UI.
+    // When there are no project-wide actions asset, the InputSystemProvider should still work as it currently gets
+    // the actions from DefaultActionsAsset().asset.
+
     [Test]
     [Category(kTestCategory)]
-    // Presses a gamepad left stick left and verifies that a navigation move event is dispatched
-    public void NavigationMoveWorks()
+    [TestCase(true)]
+    [TestCase(false)]
+    public void UI_Input_NavigationActionsWork(bool useProjectWideActionsAsset)
```

Comment:
What does "work" mean? I would suggest renaming this test to provide more clarity, e.g. "NavigationActionFiresUINavigationEvent_RegardlessIfInputIsGamepadJoystickOrKeyboard"